sdk/client/src/move_deserialize.rs

```diff
@@ -0,0 +1,69 @@
+// Copyright (c) The Diem Core Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::{Error, Result};
+use diem_json_rpc_types::views::{AccountStateWithProofView, EventWithProofView};
+use diem_types::{
+    account_state::AccountState,
+    account_state_blob::AccountStateBlob,
+    contract_event::{ContractEvent, EventWithProof},
+};
+use move_core_types::move_resource::MoveResource;
+use serde::de::DeserializeOwned;
+use std::convert::TryFrom;
+
+/// Wrapper for a deserialized Move event and its containing `ContractEvent`
+#[derive(Debug, Clone)]
+pub struct Event<T: MoveResource + DeserializeOwned> {
+    /// The deserialized event type
+    data: T,
+    event: ContractEvent,
+}
+
+/// Deserialize and return the Move events of type `T` in `events`
+pub fn get_events<T: MoveResource + DeserializeOwned>(
```

Comment:
Out of curiosity what's the value of limiting this function to types of `MoveResource`? Just to make it harder to misuse, so that you don't try to get something that doesn't look like a Resource?

Reply:
More or less avoiding misuse. At least today, all of the Rust wrappers for on-chain resources and events (e.g., https://github.com/diem/diem/tree/main/types/src/account_config/events) implement `MoveResource`), so this trait bound keeps you from accidentally trying to deserialize a random Rust struct.

In the future, I'd like to autogenerate these Rust wrappers from their Move implementation instead of hand-rolling them like we do today. As part of the autogen scheme, we might want to evaluate whether the `MoveResource` trait makes sense (IIUC, its only purpose is to reduce the boilerplate required when writing a wrapper by hand).